src/modals/SendModal/CheckTransaction.tsx

```diff
@@ -56,9 +57,7 @@ const SendModalCheckTransaction = ({ data, fees, onSend, onCancel }: SendModalCh
         <ModalFooterButton secondary onClick={onCancel}>
           Cancel
         </ModalFooterButton>
-        <ModalFooterButton onClick={onSend} disabled={!isSendButtonActive}>
```

Comment:
So this was a particular thing I had to think about a bit.

Consider the empty truth table below:

|                      |fromAddress|toAddress|amount|gasAmount|contractCode|contractState|script|issueTokenAmount|
|----------------------|-----------|---------|------|---------|------------|-------------|------|----------------|
|isSendButtonActive    |           |         |      |         |            |             |      |                |

Given all these parameters - when should `isSendButtonActive` be true?

All that's actually required is:
* `fromAddress`, `gasAmount`, and `script` for script case.
* `fromAddress`, `gasAmount`, `amount` for contract case.
* `fromAddress`, `toAddress`, `amount` for regular case.

We do validation on the TransactionForm view more or less already.

It's also why I renamed it from "Info Check" to "Review" - the user must review what they've entered, but what they enter may very well be valid based on a combination of parameters.

So I considered the best thing is to pass this to the accepting public node in the end if there is anything actually wrong, but for anything else, it's pretty much impossible for us to account for all user intent now with the introduction of contracts and scripts.

Reply:
> We do validation on the TransactionForm view more or less already.

That's a good point. I just want to avoid enabling buttons when there is missing info.

> All that's actually required is:
>
> - fromAddress, gasAmount, and script for script case.
> - fromAddress, gasAmount, amount for contract case.
> - fromAddress, toAddress, amount for regular case.

Well, that's easy to code then, isn't it?

```js
const isSendButtonActive = (fromAddress && gasAmount && script) || (fromAddress && gasAmount && amount) || (fromAddress && toAddress && amount)
```

Anyway, I'll verify this when I do my local tests.

Reply:
Don't spend too much energy on the Send modal tuning. I have the feeling that we'll build a separated dedicated modal for contract related transactions.

Reply:
> I just want to avoid enabling buttons when there is missing info.

100% - but I think this makes more sense on the TransactionForm, where it can give actual feedback too!

> Don't spend too much energy on the Send modal tuning

Basically my thoughts when implementing - the design is not final in any way.